README.md

```diff
@@ -23,8 +23,52 @@
 
 ## Overview
 
-Coming soon...
+High Level Architecture
 
+<div align="center">
+    <img src="static/architecture.png" alt="Architecture" height="450">
+    </div>
+
+This diagram illustrates a high-level architecture for a web framework in Mojo, designed with scalability, quick development, and developer-friendliness in mind. The architecture follows a layered approach, with each layer responsible for specific functionalities, making it easy to integrate with various components and request frameworks.
+
+## 1. Gateway Layer
+- This layer serves as the entry point for the web application.
+- WSGI (Web Server Gateway Interface) is the standard interface used for communication between web servers and web applications.
+
+## 2. Middleware Layer
+- This layer consists of multiple middlewares (Middleware 1, Middleware 2, ..., Middleware N) basically a stack of function calls.
+
+- Each Request will go through the middleware stack from top to bottom before being passed to Routing Layer and all responses will go through the same middleware stack from bottom to top.
+
+- Middleware components are responsible for handling cross-cutting concerns, such as authentication, logging, caching, and other common functionalities.
+- They can be easily plugged in or removed based on the application's requirements.
+- These Middlewares are order sensitive 
+
+## 3. Routing Layer
+- This layer contains a Router component responsible for mapping incoming requests to the appropriate handlers or views.
+- The Router analyzes the request and delegates it to the corresponding view or controller.
+
+## 4. View Layer
```

Comment:
I wonder about the view layer. To start small, we should probably pick one type of view (e.g. REST) and stick to it. Once we get a minimal first version working, we can think about supporting more types

Reply:
Yes that is the plan, we will make a base view trait that all realms (rest, template ) will have to follow... And work on rest framework for now